File: reverseString.js
const reverseString = (input) => {
    const stringArray = input.split('');
    console.log(stringArray);

    for (let i = 0; i < Math.floor(stringArray.length / 2); i++) {
        let temp = input[i];
        stringArray[i] = stringArray[stringArray.length - i - 1];
        stringArray[stringArray.length - i - 1] = temp;
    }
    console.log(stringArray);

    return stringArray.join('');
};

const reverseWord = (input) => {
    const stringArray = input.split(' ');
    console.log(stringArray);

    for (let i = 0; i < Math.floor(stringArray.length / 2); i++) {
        let temp = stringArray[i];
        stringArray[i] = stringArray[stringArray.length - i - 1];
        stringArray[stringArray.length - i - 1] = temp;
    }
    console.log(stringArray);

    return stringArray.join(' ');
};

// console.log(reverseString('Mary'));
console.log(reverseWord(`Deji's is going to the market`));

function reverseStringUsingStack() {
    function Stack() {
        this.items = [];
        
        this.empty = function () {
            return this.items.length === 0;
        }
        
        this.add = function(item) {
            this.items.push(item);
        }
        
        this.peek = function(item) {
            return this.items[this.items.length - 1];
        }

        this.pop = function(item) {
            return this.items.pop();
        }
    }
    
    var ans = new Stack();
    var s = '' 
    
    for (var i = 0; i < A.length; i++) {
        // console.log(A[i])
        if (A[i] === ' ') {
            ans.add(s)
            s = ''
        } else {
            s += A[i]
        }
    }
    ans.add(s)

    
    
    var result = ''
    while(!ans.empty()) {
        result += ans.pop()
        if (ans.items.length > 0) {
            result += ' '
        }
    }
    
    return result
}

File: reverseString.py


def reverse_string(s):
    chars = list(s)
    print('initial array {}'.format(chars))
    for i in range(len(s) // 2):
        chars[i], chars[len(s) - i - 1] = chars[len(s) - i - 1], chars[i]
        print(chars)
    return ''.join(chars)

def reverse_word(s):
    chars = s.split(' ')
    print(chars)
    for i in range(len(chars) // 2):
        chars[i], chars[len(chars) - i - 1] = chars[len(chars) - i - 1], chars[i]
    return ' '.join(chars)


if __name__ == '__main__':
    print(reverse_string('market'))
    # print(reverse_word('deji is going to the market'))